core/cache/application.go

```diff
@@ -54,6 +54,9 @@ func (a *Application) Config() map[string]interface{} {
 }
 
 // WatchConfig creates a watcher for the application config.
+// Do not use this to watch config on behalf of a unit.
+// It is not aware of branch-based config deltas
+// and only deals with master settings.
 func (a *Application) WatchConfig(keys ...string) *ConfigWatcher {
```

Comment:
I left this in, despite not having a current use-case. The config watcher functionality is all generalised anyway, so it isn't much code, and it doesn't pollute the intent of other functionality.